### PR TITLE
47224_migration

### PR DESCRIPTION
**Pull request by asifbuetcse**

### CODE REVIEW

**Comment by Omid07** [comment]
ok

**Comment by yasin-buet** [comment]
ok

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -17,6 +17,10 @@ public function up()
             $table->string('name');
             $table->string('email')->unique();
             $table->string('password', 60);
+            $table->string('avatar');
+            $table->boolean('role');
+            $table->UnsignedInteger('course_id')->nullable();
```
Comment (dinhhoanglong91) [inline database/migrations/2014_10_12_000000_create_users_table.php]: Sorry, my fault not checking DB certainly.
Although at a time, user can only be assigned to one course, you should store information of previous course (start time, end time,...). So, the relationship between users and course is many to many and you need `course_user` table.
In that table, you should store `course_id`, `user_id` and status: `done`, or `doing`,... Time-related information also should be saved.

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 2]
```diff
@@ -17,6 +17,10 @@ public function up()
...
+            $table->boolean('is_enrolled');
```
Comment (dinhhoanglong91) [inline database/migrations/2014_10_12_000000_create_users_table.php]: If you have table `course_user`, this column is no longer required.

File: database/migrations/2015_12_01_070413_relation_users_table.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function(Blueprint $table) {
+            $table->foreign('course_id')->references('id')->on('courses')->onDelete('cascade');
```
Comment (dinhhoanglong91) [inline database/migrations/2015_12_01_070413_relation_users_table.php]: If you have table course_user, this foreign key is no longer required.

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (asifbuetcse) [inline database/migrations/2014_10_12_000000_create_users_table.php]: Ok; so I misunderstood the requirement; apparently. 
I am changing the design now to meet with the requirement.

**Comment by Omid07** [comment]
ok

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**